GitCommands/ExternalLinks/ExternalLinksManager.cs

```diff
@@ -2,26 +2,56 @@
 using System.Collections.Generic;
 using System.Linq;
 using GitCommands.Settings;
+using GitUIPluginInterfaces;
 
 namespace GitCommands.ExternalLinks
 {
     // NB: this implementation is stateful
     public sealed class ExternalLinksManager
     {
-        private readonly RepoDistSettings _cachedSettings;
+        private readonly ISettingsSource _cachedSettings;
         private readonly ExternalLinksManager? _lowerPriority;
         private readonly IExternalLinksStorage _externalLinksStorage = new ExternalLinksStorage();
         private readonly List<ExternalLinkDefinition> _definitions;
 
-        public ExternalLinksManager(RepoDistSettings settings)
+        public ExternalLinksManager(RepoDistSettingsSet settingsSet, SettingLevel settingLevel, bool cascade = false)
         {
-            _cachedSettings = new RepoDistSettings(null, settings.SettingsCache, settings.SettingLevel);
-            _definitions = _externalLinksStorage.Load(_cachedSettings).ToList();
+            if (settingLevel is SettingLevel.Effective && !cascade)
+            {
+                throw new InvalidOperationException("Effective settings level should be cascading.");
+            }
```

Comment:
Is there any call site which is not setting `cascade: CurrentSettings.SettingLevel is SettingLevel.Effective`?

Reply:
![image](https://user-images.githubusercontent.com/3169265/117352751-04a3ef00-aeb8-11eb-88bb-b6aaeb53d158.png)

Reply:
Tests do not count. Is there use case which will be implemented in the near future?

Reply:
As far as I understood the question, no.
For the current implementation, `effectiveSettings: SettingLevel.Effective` with `cascade: false` is an exception.

Reply:
Since the `cascade` argument is specified independently of `SettingLevel.Effective` only in tests, I suggest to remove this redundant argument.